docs/ai-core-APIs.md

```diff
@@ -0,0 +1,195 @@
+# Available APIs
+
+last updated: 24.07.2024
+
+✅ Available and tested
+
+❗Warning: The APIs not marked as tested in this section are in an experimental state. Use them at your own risk.
+
+# AI Core
+
+- ApplicationAPI
+- ArtifactApi ✅
+- ConfigurationApi ✅
+- DeploymentApi ✅
+- DockerRegistrySecretApi
+- ExecutableApi
+- ExecutionApi ✅
+- ExecutionScheduleApi
+- FileApi
+- KpiApi
+- MetaApi
+- MetricsApi
+- ObjectStoreSecretApi
+- RepositoryApi
+- ResourceApi
+- ResourceGroupApi
+- ResourceQuotaApi
+- ScenarioApi ✅
+- SecretApi
+- ServiceApi
+
+## ArtifactApi
+
+### Methods
+- [artifactQuery](../packages/ai-core/src/artifact-api.ts#L29)
+- [artifactCreate](../packages/ai-core/src/artifact-api.ts#L54)
+- [artifactGet](../packages/ai-core/src/artifact-api.ts#L73)
+- [artifactCount](../packages/ai-core/src/artifact-api.ts#L92)
+
+Some use case examples are listed below.
+
+### Create an Artifact
+```TypeScript
+async function createArtifact() {
+    
+    const requestBody: ArtifactPostData = {
+        name: 'training-test-dataset',
+        kind: 'dataset',
+        url: 'https://ai.example.com',
+        scenarioId: 'foundation-models'
+    }
+
+    const responseData: ArtifactCreationResponse = await ArtifactApi
+        .artifactCreate(requestBody, {'AI-Resource-Group': 'default'})
+        .execute(destination);
+    
+    if (!responseData) {
+        throw new Error("Artifact creation failed: received null or undefined response");
+    }
```

Comment:
Arguably this should never happen, the server guarantees a 201 response with content or a 400 error.